pkg/build/builder/cmd/scmauth/password.go

```diff
@@ -114,6 +120,15 @@ func doSetup(sourceURL url.URL, usernameSecret, passwordSecret, tokenSecret stri
 		username = DefaultUsername
 	}
 
+	// If password/token is longer than what curl can handle, use a proxy
+	if len(password) > curlPasswordThreshold {
+		localURL, err := startProxy(baseDir, &sourceURL, username, password)
```

Comment:
the intent of this function was to make it very unit testable... keeping file reads/writes outside. starting a proxy in the middle of it is going to interfere with that